lib/action_view/component/base.rb

```diff
@@ -231,6 +234,8 @@ def type
           "text/html"
         end
       end
+
+      ActiveSupport.run_load_hooks(:action_view_component, self)
```

Comment:
Do we need to do this and the include above in production?

Reply:
I'm not sure, I'll double check

Reply:
1. The inclusion of ActionView::Component::Previews it's just a module that adds module attribute accessors for `preview_path` and `show_previews`. No more than that. So no harm in there, and we'll actually need them in production, in case the application has been configured to display the previews.

2. If we do not have this `ActiveSupport.run_load_hooks(:action_view_component, self)` we won't get our ActionView::Component configured with the options from the environment config file. Hence routes won't be defined. (This is what I gathered from trial and error testing, and reading the code, so I'm not entirely sure if this answers your question correctly)

Reply:
That makes sense. Perhaps we could put this line behind a conditional that checks to see if previews are enabled for the current environment?

Reply:
Yep, good idea

Reply:
Do you have any ideas on how to approach this? 

I couldn't find a way cause it seems like this class is interpreted before rails is initialized.

This is from ActionMailer::Base https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer/base.rb#L1035

Reply:
Oh! Let's just do what ActionMailer does. ☝️  Sorry for the trouble.

Reply:
It's ok no prob